apis/k8s/builder_hpa.go

```diff
@@ -126,6 +154,45 @@ func buildHPA(obj interface{}, groupVersionKind GroupVersionKind) (HPA, error) {
 	}
 }
 
+func buildHPAV2(hpa *v2.HorizontalPodAutoscaler) HPA {
+	var targetCPUPercentage int32 = 0
+	var targetMemoryPercentage int32 = 0
+	netrics := hpa.Spec.Metrics
+	for i := 0; i < len(netrics); i++ {
+		metric := netrics[i]
+		if metric.Type == v2.ResourceMetricSourceType {
+			res := metric.Resource
+			target := res.Target
+			if res.Name == "cpu" && target.AverageUtilization != (*int32)(nil) {
+				targetCPUPercentage = *target.AverageUtilization
+			} else if res.Name == "memory" && target.AverageUtilization != (*int32)(nil) {
+				targetMemoryPercentage = *target.AverageUtilization
+			}
```

Comment:
With more people using other metrics, shouldn't we do something if the metric is neither CPU or Memory? This would require update other beta versions as well. WDYT?

Reply:
Hi @fernandorubbo , my apologies I completely meant to add this PR to our fork, and must have messed that up. Happy to help get this merged upstream though if you like.

> Do you use this for getting VPA recommendations, or only for HPA? Do you use GKE? I'm asking because in GKE you have all VPA recommendations (for Deployments) in Cloud Monitoring even without deploying any VPA at all

It's a good question, we kind-of-not-really use this for the bin packing dashboard, [specifically from this article](https://cloud.google.com/kubernetes-engine/docs/tutorials/right-size-workloads-at-scale#overview). However, as you mentioned, VPA recommendations are now available in GKE, at least recently from this year, so maybe this is not even worth updating, and we would better suited to deprecate this. I'm not sure how useful the bin packing dashboard really even is now that I look at it for the first time in a few months.

The driver behind this PR was the migration warning for going to k8s 1.26, which I tracked down to the metrics exporter in this repo.
<img width="1026" alt="Screenshot 2023-10-18 at 4 26 59 PM" src="https://github.com/GoogleCloudPlatform/gke-cost-optimization-monitoring/assets/739932/96a87934-957a-44d3-a683-510259512a19">

WDYT?

Reply:
I do think this is useful for getting HPA data (yet not available in Cloud Monitoring). So, if you have tested it well enough I'll be happy to accept this PR.

Regarding to the tutorial, we removed the old one and pointed to this new one. This new one is more focused on right sizing. And I highly recommend to implement it into your company. We are getting feedback if users find this useful to integrate or not into the GKE UI somehow

Reply:
Apologies I was on vacation, let me take a look this week.